backend/app/deployment_handler.go

```diff
@@ -315,6 +314,18 @@ func (h *Handler) HandleDeployCluster(c *gin.Context) {
 		return
 	}
 
+	if err := cluster.Validate(); err != nil {
+		Error(c, http.StatusBadRequest, "Validation failed", err.Error())
+		return
+	}
+
+	projectName := kubedeployer.GetProjectName(config.UserID, cluster.Name)
+	_, err = h.db.GetClusterByName(config.UserID, projectName)
+	if err == nil {
+		c.JSON(http.StatusConflict, gin.H{"error": "deployment already exists"})
+		return
+	}
+
 	wfName := fmt.Sprintf("deploy_%d_nodes_%s", len(cluster.Nodes), config.UserID)
```

Comment:
can we add check for db error?

```suggestion
	projectName := kubedeployer.GetProjectName(config.UserID, cluster.Name)
	_, err = h.db.GetClusterByName(config.UserID, projectName)
	if err == nil {
		c.JSON(http.StatusConflict, gin.H{"error": "deployment already exists"})
		return
	}
    if !errors.Is(err, gorm.ErrRecordNotFound) {
    c.JSON(http.StatusInternalServerError, gin.H{"error": "database error"})
    return
    }
	wfName := fmt.Sprintf("deploy_%d_nodes_%s", len(cluster.Nodes), config.UserID)
```